samcli/cli/context.py

```diff
@@ -144,6 +147,11 @@ def _refresh_session(self):
         """
         try:
             boto3.setup_default_session(region_name=self._aws_region, profile_name=self._aws_profile)
+            # get botocore session and setup caching for MFA based
+            boto3.DEFAULT_SESSION._session.get_component("credential_provider").get_provider(  # pylint: disable=W0212
```

Comment:
This is using a private member. We should find a way not to rely on this. It boto changes this at all, we break.